### ISSUE TITLE
Agent MQTT Connections fail to authenticate due to invalid credentials

### ISSUE THREAD

**Issue by krital** [netdata/netdata#10825]
##### OS / Environment
Any
```

```

##### Netdata version
Unknown
 

##### Component Name
ACLK

##### Steps To Reproduce
Check the production logs using this query: 

https://console.cloud.google.com/logs/query;query=resource.type%3D%22k8s_container%22%20resource.labels.cluster_name%3D%22production%22%20resource.labels.namespace_name%3D%22infra%22%20resource.labels.container_name%3D%22vernemq%22%20textPayload:%2528%22invalid%22%2529;timeRange=P30D?project=netdata-cloud-prod-200202

Our cloud log retention is 30 days and it seems that a month ago we had a big spike (over 1M daily) but after Feb 24 the number is pretty small, yet it happens almost daily. I suspect the new behaviour for stopping MQTT session take over results in agents still trying to connect to MQTT but using invalid credentials.

##### Expected behavior
An agent that fails the challenge/response should NOT attempt to connect to MQTT.

**Comment by stelfrag** [comment]
There shouldn't be agents  trying to connect with wrong credentials unless by the time the challenge / response succeeds and returns a valid password the information has been removed from the database.

**Comment by krital** [comment]
What about if an agent is connected and another agent tries to connect with the same claim id? This used to result in MQTT session takeover but now we stop it at challenge / response? My suspicion was that maybe the agent still tries to connect after a failed challenge / response?

**Comment by krital** [comment]
I did a similar exercise like i did for the other issues, got the last 10K log entries from cloud as CSV, created a csv with timestamp, clientID and source which i attach 
[invalid.csv.zip](https://github.com/netdata/netdata/files/6184616/invalid.csv.zip)

In there we can see a lot of repeated clientIDs which are valid claim IDs (expected) but also a lot of anon-xyz client IDs as well as some others like salt. These are probably hacking attempts, so i have filtered them out from the distinct list which is only 181 claim IDs long, i also attach 
[invalid-distinct-filtered.csv.zip](https://github.com/netdata/netdata/files/6184639/invalid-distinct-filtered.csv.zip)

I will try to find out if there is any version pattern here and come back with the results.

**Comment by stelfrag** [comment]
> What about if an agent is connected and another agent tries to connect with the same claim id? This used to result in MQTT session takeover but now we stop it at challenge / response? My suspicion was that maybe the agent still tries to connect after a failed challenge / response?

The `legacy` does not because it fails to find a password after a failed challenge / response. I see the NG setting `anon` there.

**Comment by underhood** [comment]
@stelfrag yep I see NG can attempt connection if it failed OTP. Will fix that

**Comment by stelfrag** [comment]
> @stelfrag yep I see NG can attempt connection if it failed OTP. Will fix that. But currently there are but few agents running NG in the wilderness.

I agree -- NG doesn't explain the big numbers here
